packages/nc-gui/composables/useViewFilters.ts

```diff
@@ -425,6 +432,37 @@ export function useViewFilters(
     },
   )
 
+  // method to extract looked up column meta for all bt lookup columns
+  // it helps to decide the condition operations for the column
+  const loadBtLookupTypes = async () => {
+    const btLookupTypes = {}
+    try {
+      for (const col of meta.value?.columns || []) {
+        if (col.uidt !== UITypes.Lookup) continue
+        let nextCol = col
+        let btLookup = true
+        // check all the relation of nested lookup columns is bt or not
+        // include the column only if all only if all relations are bt
+        while (btLookup && nextCol && nextCol.uidt === UITypes.Lookup) {
```

Comment:
If it ends up like that it will cause problem internally as well, will check that part. To avoid this I will check at any point it's referring the same column, if yes I will break the loop.

Reply:
<img width="730" alt="image" src="https://github.com/nocodb/nocodb/assets/61551451/bd1609fd-394a-422b-ad1b-8dd5a868f357">